packages/main-api/src/controllers/v1/teams/groups.rs

```diff
@@ -126,6 +153,20 @@ impl GroupController {
             let _ = GroupMember::get_repository(self.pool.clone())
                 .insert_with_tx(&mut *tx, user_id, id)
                 .await?;
+
+            // Send invitation notification to the user
+            let notification_data = NotificationData::InviteTeam {
+                team_id,
+                group_id: id,
+                image_url: Some(team.profile_url.clone()),
+                description: format!("You have been invited to join {} in {}", group.name, team.nickname),
+            };
+
+            // Send notification outside of the transaction to avoid potential deadlocks
+            if let Err(e) = send_notification(&self.pool, auth_for_notifications.clone(), user_id, notification_data).await {
```

Comment:
send_notification causes unneccessary overhead for querying user_id. 
And it doesn't support transation.